core/services/gateway/handlers/functions/allowlist.go

```diff
@@ -172,17 +202,48 @@ func (a *onchainAllowlist) updateFromContractV1(ctx context.Context, blockNum *b
 	if err != nil {
 		return errors.Wrap(err, "unexpected error during functions_allow_list.NewTermsOfServiceAllowList")
 	}
-	addrList, err := tosContract.GetAllAllowedSenders(&bind.CallOpts{
```

Comment:
By removing this we are introducing a deployment order dependency (can't deploy new node version before new ToS contract s deployed). Let's put the new fetching API under a feature flag in config to avoid that. Persisting in DB can be enabled by default.